setup.py

```diff
@@ -0,0 +1,16 @@
+from setuptools import setup
+
+with open("requirements.txt") as requirements:
+    requirements = requirements.readlines()
+
+setup(
+    name='versations',
+    version='0.1.0',
+    py_modules=['main'],
+    install_requires=requirements,
```

Comment:
you are then restricting yourself to not be able to use `requirements.txt` for exact pinning of versions etc, since here you really do not want to pin but to establish sensible minimal (`>=`) or "release" (`~=`) dependencies.  

I think it might have been https://caremad.io/posts/2013/07/setup-vs-requirement/ which inspired me to be clear on such separation.  And quite often I do everything in setup.py and then in `requirements.txt` have just 

```
-e .[devel]
```
for devel (e.g. https://github.com/datalad/datalad/blob/maint/requirements-devel.txt)  or 

```
.[full]
```
for a regular one (https://github.com/datalad/datalad/blob/maint/requirements.txt).  But I am ok if someone really wants tightly pinned on for some specific deployment e.g. of service we use as @candleindark has in https://github.com/datalad/datalad-registry/blob/master/requirements.txt .

PS note that these days better to specify all of those in `setup.cfg` not `setup.py`.  See e.g. https://github.com/con/tinuous/blob/master/setup.cfg#L39